frontend/app/directives.js

```diff
@@ -13,30 +13,134 @@ angular.module('TicTacToe.directives', [])
                 }
             }
         }
     }])
-    .directive('singleBoard', ['$http', 'gameService', function($http, gameService) {
+    .directive('boardHighlight', ['tictactoe', function(tictactoe) {
+        function linker(scope, element, attrs) {
+            var $el = $(element);
+            scope.curHighlightRow = undefined;
+            scope.curHighlightCol = undefined;
+
+            var setHighlight = function(boardRow, boardCol, cellRow, cellCol) {
+                var pos = $el.position();
+
+                scope.curHighlightRow = boardRow;
+                scope.curHighlightCol = boardCol;
+                console.log(boardRow, boardCol, cellRow, cellCol);
+
+                return new Promise(function(resolve, reject) {
+                    var $board = $(".small-board.row-$ROW.column-$COL".replace("$ROW", boardRow+1).replace("$COL", boardCol+1));
+                    if (typeof cellRow === "undefined" || typeof cellCol === "undefined") {
+                        console.log("Highlight Board:", boardRow, boardCol);
+
+                        var isClosed = $board.hasClass('playerOne') || $board.hasClass('playerTwo');
+                        var afterAnimation;
+                        if (isClosed) {
+                            afterAnimation = function() {
+                                var ultimateBoard = $('.ultimate-board');
+                                tictactoe.highlightPulse().animate({
+                                    top: 0,
+                                    left: 0,
+                                    width: ultimateBoard.width(),
+                                    height: ultimateBoard.height(),
+                                }, 700, resolve);
+                            };
+                        } else {
+                            afterAnimation = resolve;
+                        }
+                        $el.animate({
+                            top: 312 * boardRow,
+                            left: 312 * boardCol,
+                            width: $board.outerWidth(),
+                            height: $board.outerWidth(),
+                            borderRadius: 15,
+                        }, 500, "swing", afterAnimation);
+                    } else {
+                        console.log("Highlight Cell:", cellRow, cellCol);
+                        var firstTop = 7 + 312 * boardRow + 98 * cellRow;
+                        var firstLeft = 7 + 312 * boardCol + 98 * cellCol;
+                        $el.animate({
+                            top: firstTop,
+                            left: firstLeft,
+                            width: 99, height: 99,
+                            borderRadius: 50,
+                        }, 500, "swing", resolve);
+                    }
+                })
+            }
+            tictactoe.setHighlight = setHighlight;
```

Comment:
well, you could move it to the tictactoe factory itself (or create a new one), and just pass the element to it. I mean, we could file a separate refactor ticket for that if you want so that we can merge this PR.